src/Radio/Radio.js

```diff
@@ -0,0 +1,293 @@
+              },
+
+              '&:hover,&[data-simulate-hover]': {
+                '& + span': {
+                  backgroundColor:
+                    theme.RadioControl_backgroundColor_checkedHover,
+                  borderColor: theme.RadioControl_borderColor_checkedHover
+                }
+              },
+
+              '&:focus,&[data-simulate-focus]': {
+                '& + span': {
+                  borderColor: theme.RadioControl_borderColor_checkedFocus,
+                  boxShadow: theme.RadioControl_boxShadow_checkedFocus
+                }
+              }
+            })
+      }
+    };
+  },
+  root: ({ disabled, theme: baseTheme }) => {
+    const theme = componentTheme(baseTheme);
+
+    return {
+      alignItems: 'center',
+      cursor: !disabled && 'pointer',
+      display: 'flex',
+      position: 'relative',
+
+      '&:hover,&[data-simulate-hover]': {
+        '& span:first-of-type': {
+          borderColor: !disabled && theme.RadioControl_borderColor_hover
+        }
+      }
+    };
+  },
+  text: ({ disabled, size, theme: baseTheme }) => {
+    const theme = componentTheme(baseTheme);
+
+    return {
+      color: disabled ? theme.color_text_disabled : theme.RadioText_color_text,
+      fontSize:
+        size === 'small'
+          ? theme.RadioText_fontSize_small
+          : theme.RadioText_fontSize
+    };
+  }
+};
+
+const Root = createStyledComponent('label', styles.root, {
+  displayName: 'Radio',
+  includeStyleReset: true
+});
+const Input = createStyledComponent('input', styles.input, {
+  rootEl: 'input'
+});
+const Text = createStyledComponent('span', styles.text);
+const Control = createStyledComponent('span', styles.control);
+
+/**
+ * Radio allows users to select a single option from a set.
+ *
+ * Mineral UI also provides a [RadioGroup](../radio-group), which offers a
+ * simpler API for working with a group of Radios, and a
+ * [FormField](../form-field), which adds labeling, messaging, and additional
+ * accessibility.
+ */
+export default function Radio({
+  className,
+  disabled,
+  inputRef,
+  invalid,
+  label,
+  required,
+  rootProps: otherRootProps,
+  size = 'large',
+  ...restProps
+}: Props) {
+  const rootProps = {
+    className,
+    disabled,
+    ...otherRootProps
+  };
+
+  const inputProps = {
+    'aria-invalid': invalid,
+    'aria-required': required,
+    disabled,
+    innerRef: ref => {
+      if (inputRef) {
+        inputRef(ref);
+      }
+    },
+    required,
+    size,
+    type: 'radio',
+    ...restProps // Note: Props are spread to Input rather than Root
+  };
+
+  const controlProps = {
+    disabled,
+    size
+  };
+
+  const textProps = {
+    disabled,
+    size
+  };
+
+  return (
+    <Root {...rootProps}>
+      <Input {...inputProps} />
+      <Control {...controlProps}>
+        <svg viewBox="0 0 24 24" role="img" aria-hidden="true">
+          <circle cx="12" cy="12" r="6" />
```

Comment:
no animation??

Reply:
I would like to see transitions as well, but that can come post-MVP when a more considered approach can be taken.